File: typescript/tsfind/src/sortby.ts
/*
 * filetype.ts
 *
 * FileType enum
 *
 */

'use strict';

export enum SortBy {
    FilePath,
    FileName,
    FileSize,
    FileType,
    LastMod
}

export function nameToSortBy(name: string): SortBy {
    const uname: string = name.toUpperCase();
    if (uname === 'NAME')
        return SortBy.FileName;
    if (uname === 'SIZE')
        return SortBy.FileSize;
    if (uname === 'TYPE')
        return SortBy.FileType;
    if (uname === 'LASTMOD')
        return SortBy.LastMod;
    return SortBy.FilePath;
}

export function sortByToName(sortBy: SortBy): string {
    if (sortBy === SortBy.FileName)
        return 'NAME';
    if (sortBy === SortBy.FileSize)
        return 'SIZE';
    if (sortBy === SortBy.FileType)
        return 'TYPE';
    if (sortBy === SortBy.LastMod)
        return 'LASTMOD';
    return 'PATH';
}

File: java/javafind/src/main/java/javafind/SortBy.java
package javafind;

public enum SortBy {
    FILEPATH("filepath"),
    FILENAME("filename"),
    FILESIZE("filesize"),
    FILETYPE("filetype"),
    LASTMOD("lastmod");

    private final String name;

    private SortBy(final String name) {
        this.name = name;
    }

    public static SortBy forName(final String name) {
        String lname = name.trim().toLowerCase();
        return switch (lname) {
            case "filename", "name" -> FILENAME;
            case "filesize", "size" -> FILESIZE;
            case "filetype", "type" -> FILETYPE;
            case "lastmod" -> LASTMOD;
            default -> FILEPATH;
        };
    }

    public String toName() {
        return this.name;
    }
}
